### PR TITLE
etcdctl: add migrate command into readme

### PR DESCRIPTION
**Pull request by xiang90**
@heyitsanthony @hongchaodeng @gyuho @joshix

### CODE REVIEW

File: etcdctl/READMEv3.md [hunk 1]
```diff
@@ -735,6 +735,59 @@ cf1550fb, 3, 3, 25 kB
...
+#### User-provided transformer
+
+Users can provide a customized transformer function that transforms a key from v2 store to any number of keys in mvcc store. Migration program writes JSON format of the [v2 store key][v2key] into the stdin of the transformer program, reads protobuf format of [mvcc key][v3key] back from the stdout of the transformer program, and save the transformed keys into mvcc store to finish migration.
+
+The provided transformer should read until EOF and flush the stdout before exiting to ensure data integrity.
+
+#### Example
+
+```
+./etcdctl --data-dir=/var/etcd --transformer=k8s-transformer
```
Comment (gyuho) [inline etcdctl/READMEv3.md]: `ETCDCTL_API=3` to clarify that this belong to v3 ctl?

Comment (gyuho) [inline etcdctl/READMEv3.md]: Can we have a simple example of customized transformer?

Comment (xiang90) [inline etcdctl/READMEv3.md]: This is etcdctl3 docs. We do not add the env for all commands listed in this doc.

**Comment by gyuho** [comment]
lgtm. defer to other team members

File: etcdctl/READMEv3.md (hunk 1 above)
Comment (gyuho) [inline etcdctl/READMEv3.md]: right. nvm.

File: etcdctl/READMEv3.md [hunk 2]
```diff
@@ -735,6 +735,59 @@ cf1550fb, 3, 3, 25 kB
...
+		ModRevision:    int64(n.ModifiedIndex),
+		Version:        1,
+	}
+	return kv
+}
+```
+
+#### User-provided transformer
+
+Users can provide a customized transformer function that transforms a key from v2 store to any number of keys in mvcc store. Migration program writes JSON format of the [v2 store key][v2key] into the stdin of the transformer program, reads protobuf format of [mvcc key][v3key] back from the stdout of the transformer program, and save the transformed keys into mvcc store to finish migration.
```
Comment (chzchzchz) [inline etcdctl/READMEv3.md]: Users can provide a customized 1:n transformer function that transforms a key from the v2 store to any number of keys in the mvcc store. The migration program writes JSON formatted [v2 store keys][v2key] to the transformer program's stdin, reads protobuf formatted [mvcc keys][v3key] back from the transformer program's stdout, and finishes migration by saving the transformed keys into the mvcc store.

Comment (heyitsanthony) [inline etcdctl/READMEv3.md]: Users can provide a customized 1:n transformer function that transforms a key from the v2 store to any number of keys in the mvcc store. The migration program writes JSON formatted [v2 store keys][v2key] to the transformer program's stdin, reads protobuf formatted [mvcc keys][v3key] back from the transformer program's stdout, and finishes migration by saving the transformed keys into the mvcc store.

**Comment by xiang90** [comment]
@heyitsanthony Fixed.

File: etcdctl/READMEv3.md [hunk 3]
```diff
@@ -735,6 +735,59 @@ cf1550fb, 3, 3, 25 kB
...
+
+Simple reply
+
+- Exit code is zero when migration is finished successfully.
+
+- Error string if migration failed. Exit code is non-zero.
+
+#### Default transformer
+
+If user does not provide a transformer program, migrate command will use the default transformer. The default transformer transforms keys in store v2 into mvcc store according to the following Go program:
```
Comment (heyitsanthony) [inline etcdctl/READMEv3.md]: The default transformer transforms `storev2` formatted keys into `mvcc` formatted keys according to the following Go program:

Comment (xiang90) [inline etcdctl/READMEv3.md]: fixed.

File: etcdctl/READMEv3.md [hunk 4]
```diff
@@ -735,6 +735,59 @@ cf1550fb, 3, 3, 25 kB
 +----------+----------+------------+------------+
 ```
 
+### MIGRATE [options]
+
+Migrate migrates keys in store v2 into mvcc store.
```
Comment (heyitsanthony) [inline etcdctl/READMEv3.md]: Migrate migrates keys in a v2 store to a mvcc store.

**Comment by heyitsanthony** [comment]
lgtm after final few nits

**Pull request merged**